contrib/spring/src/test/java/io/opencensus/contrib/spring/instrument/web/TraceWebAsyncClientAutoConfigurationTest.java

```diff
@@ -0,0 +1,227 @@
+  @Order(1)
+  // Need to suppress warnings for MustBeClosed because Java-6 does not support try-with-resources.
+  @SuppressWarnings("MustBeClosedChecker")
+  public void should_close_span_upon_success_callback()
+      throws ExecutionException, InterruptedException {
+    this.tracer = Tracing.getTracer();
+    Span initialSpan = this.tracer.spanBuilder("initial").startSpan();
+
+    Scope ws = null;
+    try {
+      ws = this.tracer.withSpan(initialSpan);
+      ListenableFuture<ResponseEntity<String>> future =
+          this.asyncRestTemplate.getForEntity(
+              "http://localhost:" + port() + "/async", String.class);
+      String result = future.get().getBody();
+
+      assertThat(result).isEqualTo("async");
+    } catch (Exception e) {
+      System.out.println(e);
+    } finally {
+      if (ws != null) {
+        ws.close();
+      }
+      initialSpan.end();
+    }
+
+    // 3 spans are initial, client, server.
+    List<SpanData> spans = handler.waitForExport(3);
+    SpanData clientSpan = null;
+    for (SpanData span : spans) {
+      if (span.getKind() == CLIENT) {
+        clientSpan = span;
+        assertThat(clientSpan.getName()).isEqualTo("/async");
+        assertThat(clientSpan.getStatus().isOk()).isTrue();
+        assertThat(
+                clientSpan
+                    .getAttributes()
+                    .getAttributeMap()
+                    .get(HttpTraceAttributeConstants.HTTP_METHOD))
+            .isEqualTo(AttributeValue.stringAttributeValue("GET"));
+        assertThat(
+                clientSpan
+                    .getAttributes()
+                    .getAttributeMap()
+                    .get(HttpTraceAttributeConstants.HTTP_HOST))
+            .isEqualTo(AttributeValue.stringAttributeValue("localhost"));
+        assertThat(
+                clientSpan
+                    .getAttributes()
+                    .getAttributeMap()
+                    .get(HttpTraceAttributeConstants.HTTP_PATH))
+            .isEqualTo(AttributeValue.stringAttributeValue("/async"));
+        assertThat(clientSpan.getKind()).isEqualTo(CLIENT);
+        break;
+      }
+    }
+    assertThat(clientSpan).isNotNull();
+  }
+
+  @Test(timeout = 10000)
+  @Order(2)
+  public void should_close_span_upon_failure_callback() {
+    final ListenableFuture<ResponseEntity<String>> future;
+    try {
+      future =
+          this.asyncRestTemplate.getForEntity("http://localhost:" + port() + "/fail", String.class);
+      new Thread(
+              new Runnable() {
+                public void run() {
+                  try {
+                    Thread.sleep(100);
+                  } catch (Exception e) {
+                    System.out.println("exception " + e);
+                  }
+                  future.cancel(true);
+                }
+              })
+          .start();
+
+      future.get(500, TimeUnit.MILLISECONDS);
+      assertThat(true).named("should throw an exception from the controller");
```

Comment:
why 'assertThat(true)'?

Reply:
actually I wanted equivalent of assertThrows(). I have re-implemented this. Let me know if that works.